test/passing/tests/open-closing-on-separate-line.ml.ref

```diff
@@ -357,3 +357,34 @@ let _ =
         match path uri with "" -> with_path uri "/" | _ -> uri
       )
     )
+
+(* Ptyp_open *)
+
+let _ : M.(foo * M.(bar)) = ()
+
+let _ : M.(foo) * M.(bar) = ()
+
+let _ : M.([`Foo of foo]) = ()
+
+let _ : M.N.(foo) = ()
+
+let _ :
+    M.(
+      foooooooooooooooooooooooooooooooooooooooo
+      * foooooooooooooooooooooooooooooooooooooooo) =
+  ()
+
+let _ :
+    M.(
+      [ `Foo of
+        foooooooooooooooooooooooooooooooooooooooo
+        * foooooooooooooooooooooooooooooooooooooooo ]) =
+  ()
+
+let _ : M.((foo[@attr])) = ()
+
+let _ : (M.(foo)[@attr]) = ()
+
+let _ : M.((foo[@attr] [@attr])) = ()
+
+let _ : (M.((foo[@attr]))[@attr]) = ()
```

Comment:
There's too many parentheses but this is something general about types and is unrelated to the new code. It shouldn't block this PR.